tidb-server/main.go

```diff
@@ -172,6 +171,8 @@ func main() {
 	pushMetric(*metricsAddr, time.Duration(*metricsInterval)*time.Second)
 
 	log.Error(svr.Run())
+	domain.Close()
```

Comment:
I have tested stop tidb-server with `ctrl-c` and `kill -1`. I am sure TiDB will run to this line.